### ISSUE TITLE
Enhancement: Embeds 

### ISSUE THREAD

**Issue by RedBrumbler** [tsickert/discord-webhook#9]
Would it be possible to support embeds directly, without having to use a custom json? (can you even do it like that? idk)

it would be preferred to also be compatible with file posting, but doesn't strictly have to be as you could of course just send 2 messages. Mostly just requesting this as embeds tend to look neater than direct messages, as seen in this image of me using it to post artifacts in a private discord channel

![image](https://user-images.githubusercontent.com/51795403/134566117-92bd147c-b4f2-40c0-9463-6bc205cacb75.png)

for comparison a github webhook embed:

![image](https://user-images.githubusercontent.com/51795403/134566190-67f914cc-a4df-43ee-88c0-305fa072203c.png)

I would understand that it would probably not be the same but in my opinion this could be a cool and useful addition to the action. Feel free to deny this as "too difficult" or "out of the intended scope" though!

**Comment by TrueTamashii** [comment]
I'm Charlotte and I support this message 🐸

**Comment by tsickert** [comment]
I've been wanting to implement a better embed option for a while, and I've got a few ideas on how to amek that work. I'll play around with it and let you know if/when I've got something to try out. Thanks for the suggestion!

**Comment by lAzrythl** [comment]
> I'm Charlotte and I support this message 🐸
I’m buzz and if you don’t do this your kneecaps will disappear… just coincidentally

**Comment by tsickert** [comment]
Alright, whipped something up that I seem to have working. It's still a pre-release, and there's still a bit more testing that I want to do, but feel free to try out [v4.0.0-preview7](https://github.com/marketplace/actions/discord-webhook-action?version=v4.0.0-preview7).

I updated the README with the new usage. Since the fields are provided as CLI arguments, the basic rule is that the input is the path to the field in the json.

For example:

```json
{
  "embeds": [
    "author": {
      "name": "foo"
    }
  ]
}
```

would be referenced as `embed-author-name` and so on. 

This version basically just passes the values along to the API, so you're able to do a lot more, but you may get some malformed requests here and there based on discord's API or if the embed exceeds the [embed limits](https://discord.com/developers/docs/resources/channel#embed-limits). 

The action also prints out the payload you're sending and the response from Discord, so debugging should be easier. 

As far as the original ask goes, there were some interesting restrictions. It turns out that webhooks that send files actually can't include embeds (which I thought was strange). So, I just built in a second webhook call to send the embeds that are provided with the file. It will show up as 2 messages, but it should be transparent when you're using it. I'm also just now looking at the example you showed and realizing that the file comes after the content. Right now the embeds are set to print second. If that doesn't work for you, I can add a flag to the action that could let you flip them.

Other things to note are that `fields` are left off of the embeds because they were a pain to implement and they really just seem to be bolded text over non-bolded text.

And, finally, this only currently support 1 embed. I can add the maximum supported number (10) at some point if it's needed, but it's tedious to add the rest and I wanted to get something out. 

Let me know if you run into anything that looks wrong and I'll be happy to take a look. Thanks!

**Comment by RedBrumbler** [comment]
It'd be preferred to have the file after the embed, that seems more natural to me.

Also I'm pretty sure 1 embed is enough, I doubt anyone would really send multiple (i'm probably gonna be proven wrong on this)

I'll have a try with it and see if it does what i'd want

**Comment by tsickert** [comment]
> It'd be preferred to have the file after the embed, that seems more natural to me.

Yeah, agree, especially since `content` shows up above the embed. That's the default behavior now in [v4.0.0-preview8](https://github.com/marketplace/actions/discord-webhook-action?version=v4.0.0-preview8) if you want to try that out.

**Comment by tsickert** [comment]
Released [v4.0.0](https://github.com/marketplace/actions/discord-webhook-action)! 🎉 

Going to go ahead and close this. Feel free to reopen or make a new ticket if you run into anything else. Thanks again for the feedback!
